common/dynamicconfig/collection_test.go

```diff
@@ -411,6 +411,45 @@ func (s *collectionSuite) TestGetTypedListOfStruct() {
 	})
 }
 
+func (s *collectionSuite) TestGetTypedProtoEnum() {
```

Comment:
minor.
from coverage point of view it looks ok, I prefer to unit-test the smallest unit of change, in this case - mapstructureHookProtoEnum 

for s.client.SetValue(testGetTypedPropertyKey, true) check that error is returned?

Reply:
I mostly disagree. That approach is a recipe for that are too coupled to the implementation (and often end up essentially rewriting it). The test should ideally be against the public interface of the unit (Collection), which this is. That specific function is an implementation detail, and the implementation should be able to change without the tests having to change.

Also note that this test is written as an external package (on the suggestion of another team member), specifically to enforce testing against the public interface only, so it can't even access that function.

It so happens that the public API doesn't return errors in any convenient way, it logs them. So we can't easily test for the error messages. You could ague that's a deficiency in the API, but it would be a larger project to fix.

Reply:
@dnr could we pass in a log.Logger implementation that fails the test if `Error(...)` is called?